keps/sig-node/2254-cgroup-v2/README.md

```diff
@@ -52,6 +63,185 @@ This proposal aims to:
+
+### Graduation Criteria
+
+- Alpha: Phase 1 completed and basic support for running Kubernetes on
+  a cgroups v2 host,  e2e tests coverage or have a plan for the
+  failing tests.
+  A good candidate for running cgroup v2 test is Fedora 31 that has
+  already switched to default to cgroup v2.
+
+- Beta: e2e tests coverage and performance testing.  Verify that both
+  the CPU and Memory Manager work.
+
+- GA: Assuming no negative user feedback based on production
+  experience, promote after 2 releases in beta.
+  *TBD* whether phase 2 must be implemented for GA.
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+N/A.  Not relevant to upgrades.  If the host is running with cgroup v2 then
+it will be automatically detected and used.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [ ] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name:
+  - Components depending on the feature gate:
+- [X] Other
+  - Describe the mechanism:
+    configure the hosts to use cgroup v2
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+    No, each host can be restarted to cgroup v2 separately
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+    It requires downtime of a node since it needs to be rebooted
+
+###### Does enabling the feature change any default behavior?
+
+N/A.  It must work in the same way as on cgroup v1
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+yes, it is enough to restart the node on cgroup v1
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+it should work seamlessly without any difference
+
+###### Are there any tests for feature enablement/disablement?
+
+The same E2E tests that work on cgroup v1 should work on cgroup v2
+
+### Rollout, Upgrade and Rollback Planning
+
+N/A.  Each node can be configured separately.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+N/A.  It requires a reboot to be enabled.  If the workload accesses directly the
+cgroup file system, then also the workload must be enabled for cgroup v2.
+
+###### What specific metrics should inform a rollback?
+
+Pods not being healthy.
```

Comment:
Maybe pods not getting the right resource constraints? Any way to detect that?

Reply:
One could inspect the pod cgroups. However, they will have to look at the conversion table in this KEP as some field values map directly while others don't.